### DISCUSSION TITLE
How to get personal Transaction data

### DISCUSSION THREAD

**Discussion by bimbambusse** [ib-api-reloaded/ib_async#68] [Q&A] [answered]
Im a IB_async-newbie (and Interactive Brokers newbie too), and I have problems getting transaction data on my activities done to my paper trading account at Interactive Brokers.

My connection seems to work OK, as ib.accountSummary() and ib.portfolio() return what they are supposed to.

However, the return from ib.executions() and ib.reqExecutions() is empty.

I suspect paper trading accounts don't save your personal transactions(?).  Or am I just using the wrong IB_async-method?

/Peter

PS: Please let me know if there are more appropriate fora for this, say <https://www.reddit.com/r/interactivebrokers/>

*Situation: I'm a Dane and the world highest taxation has led to an unbelievable amount of administration required for buying public stocks as a retail investor here in Denmark. Danish brokers usually have systems implemented so that every trade(!) is reported to the IT-system of the tax-autorities (who are then calculating your tax). But IB (and other non-Danish brokers) obviously don't have something like that set up. As I would like an IB-account for a number of (legit) reasons, I ultimately need to build my own "system" being able to report every trade (prices calculated in DDK via exchange rate etc) to the tax-autorities if needed. I was hoping to get a feel of how easy such a home-made system would be to develop, by making tests on a paper trading account, before I open a real account with real money (and wake the Dragon). But if a real account is necessary, so be it.*

**Comment by gnzsnz** [comment]
The activity statement in csv format will allow you to do a good amount of
analysis. It's what I personally use.

Flex reports with Ib_async on top would take it to the next level

Check the documentation for details

**Comment by mattsta** [comment, marked as answer]
> However, the return from ib.executions() and ib.reqExecutions() is empty.

Yeah, it can be somewhat confusing at first when you ask for account history and get nothing back.

IBKR gateway/tws APIs only report trades/executions for the most recent 1-2 calendar days then they stop showing up in the API execution list.

Older trade details are only available using account/portal website downloads in various formats or you can set up a web service API key for automatically fetching data: https://www.ibkrguides.com/clientportal/performanceandstatements/flex-web-service.htm / https://www.ibkrguides.com/clientportal/performanceandstatements/activityflex.htm
